components/host-dashboard/reports-section/HostDownloadsSection.js

```diff
@@ -0,0 +1,155 @@
+  return {
+    timezoneType: 'UTC', // To match the monthly host report sent by email
+    from: interval.from.tz('UTC', true).toISOString(),
+    to: interval.to.tz('UTC', true).toISOString(),
+  };
+};
+
+const HostDownloadsSection = ({ hostSlug }) => {
+  const [collectiveOptions, setCollectiveOptions] = React.useState(null);
+  const [dateInterval, setDateInterval] = React.useState(getDefaultDateInterval);
+  const variables = { hostSlug };
+  const { loading, data } = useQuery(hostReportDownloadsSectionQuery, { variables, context: API_V2_CONTEXT });
+  const host = data?.host;
+  const accountsSlugs = collectiveOptions?.map(c => c.value.slug);
+  const hostReportUrl = getHostReportURL(hostSlug, { ...dateInterval, accountsSlugs });
+  const { loading: isFetching, call: downloadCSV } = useAsyncCall(
+    () => {
+      const formatDate = d => dayjs(d).format('YYYY-MM-DD');
+      let filename = `host-${hostSlug}-transactions`;
+      if (dateInterval?.from) {
+        filename += `-${formatDate(dateInterval.from)}-${formatDate(dateInterval.to)}`;
+      }
+
+      return fetchCSVFileFromRESTService(hostReportUrl, filename);
+    },
+    { useErrorToast: true },
+  );
+
+  return (
+    <Container bg="black.50" borderRadius={8} p={3}>
+      <Grid gridTemplateColumns={['1fr', '1fr 2fr auto']} gridGap="8px">
+        <Box>
+          <StyledInputField
+            name="download-host-report-period"
+            label={
+              <FieldLabel>
+                <FormattedMessage id="Period" defaultMessage="Period" />
+              </FieldLabel>
+            }
+          >
+            {({ id }) => <PeriodFilter inputId={id} onChange={setDateInterval} value={dateInterval} />}
+          </StyledInputField>
+        </Box>
+        <Box>
+          <StyledInputField
+            name="download-host-report-collectives"
+            label={
+              <FieldLabel>
+                <FormattedMessage defaultMessage="Filter by collective" />
+              </FieldLabel>
+            }
+          >
+            {({ id }) => (
+              <CollectivePickerAsync
+                inputId={id}
+                onChange={setCollectiveOptions}
+                hostCollectiveIds={host ? [host.legacyId] : null}
+                disabled={loading}
+                isMulti
+              />
+            )}
+          </StyledInputField>
+        </Box>
+        <Flex alignItems="flex-start" pt={[16, 26]}>
+          <Link
+            href={hostReportUrl}
+            onClick={e => {
+              e.preventDefault();
+              downloadCSV();
+            }}
+          >
+            <StyledButton
+              buttonStyle="primary"
+              buttonSize="small"
+              py="7px"
+              width="100%"
+              minWidth={140}
+              loading={isFetching}
+            >
+              <FormattedMessage defaultMessage="Generate report" />
+            </StyledButton>
```

Comment:
This button has a specific color in design (#46347F). Maybe it's better to have it given that the section has a lot of purple kinda color elements but this is the only blue button; 😉  

![image](https://user-images.githubusercontent.com/12435965/135506713-d8876ef8-ffd7-4ac0-b4f5-5aba4856a6a1.png)

Reply:
As mentioned in https://github.com/opencollective/opencollective-frontend/pull/6849#discussion_r708957641:
> This is supposed to match `colors.primary`, not `colors.purple`. We'll work on theming before releasing, some parts of it we still need to figure out.

Reply:
The customized theming needs to be applied with a `CollectiveThemeProvider`

Reply:
Ah I totally forgot this comment. Makes sense. 👍🏽